main.go

```diff
@@ -358,6 +360,51 @@ func notifyWaiters(w http.ResponseWriter, r *http.Request) {
 
 func main() {
 	flag.Parse()
-	http.HandleFunc("/", handle)
-	log.Fatal(http.ListenAndServe(*flagHTTP, nil))
+
+	tmpDir, err := ensureTmpOutputDir()
+	if err != nil {
+		log.Fatalf("Failed to create temporary directory: %v", err)
+	}
+
+	defer os.RemoveAll(tmpDir)
+
+	var server http.Server
+
+	shutdown := make(chan struct{})
+	go func() {
+		sigint := make(chan os.Signal, 1)
+		signal.Notify(sigint, os.Interrupt)
+		<-sigint
+
+		log.Printf("Shutting down server...")
+
+		// Received an interrupt signal, shut down.
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
+		defer cancel()
+		err := server.Shutdown(ctx)
+		if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+			// Error from closing listeners, or context timeout:
+			log.Printf("Error at server.Shutdown: %v", err)
+		}
+		close(shutdown)
+
+		<-sigint
+		// hard exit the second ctrl-c
```

Comment:
```
// Hard exit the second ctrl-c
```
?

Reply:
I've done this pattern in other go programs. The issue is since ctrl-c is caught, in case of a bug or some strange behavior of the program, it will just stick around forever even if you keep pressing ctrl-c. If the user presses ctrl-c multiple times they are trying to say 'just shut down now', so in the case of a second ctrl-c the program obliges by exiting immediately.

The alternative for the user is to ctrl-z, and kill -9 the program.

Reply:
Nah, I mean the content of the comment. "Hard exit on the second ctrl-c" seems better than "hard exit the second ctrl-c". At least, the first character should be upper case.

Reply:
fixed